exercises/matthew_cumber/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,145 @@
+#include <iostream>
+#include <iomanip>
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <ctype.h>
+#include <regex>
+#include "UniqueWord.h"
+
+std::string PATH = ""; // Will store the path of the file to be read
+
+/** Tests if a word has already been found the increments it's count
+ * @param uniqueWords The vector of existing unique words found
+ * @param word The word that is being tested
+ * @return A boolean, true if a word already exists, false otherwise
+ */
+bool inUniqueWords(std::vector<UniqueWord> &uniqueWords, const std::string &word)
+{
+  // Loop unique words, if word exists increment it's count
+  for(auto &w : uniqueWords)
```

Comment:
Please can you change this to use the std::find function:

auto it = std::find(uniqueWords.begin(), uniqueWords.end(), word)
if (it!=uniqueWords.end()){
}

This is a similar quantity of code but it means it's easier to swap the type of uniqueWords over to one of the other STL container classes (list, map, queue etc) which may not support random access.
The potential for doing this in this specific situation isn't huge but as a piece of general practice this change may come in useful.